mapper/mapper_test.go

```diff
@@ -932,18 +935,400 @@ func TestCreateCensusDatasetLandingPage(t *testing.T) {
+					},
+					{
+						DimensionID: "Dimension 1",
+						Label:       "Label 13",
+					},
+					{
+						DimensionID: "Dimension 1",
+						Label:       "Label 14",
+					},
+					{
+						DimensionID: "Dimension 1",
+						Label:       "Label 15",
+					},
+					{
+						DimensionID: "Dimension 1",
+						Label:       "Label 16",
+					},
+					{
+						DimensionID: "Dimension 1",
+						Label:       "Label 17",
+					},
+					{
+						DimensionID: "Dimension 1",
+						Label:       "Label 18",
+					},
+					{
+						DimensionID: "Dimension 1",
+						Label:       "Label 19",
+					},
+					{
+						DimensionID: "Dimension 1",
+						Label:       "Label 20",
+					},
+					{
+						DimensionID: "Dimension 1",
+						Label:       "Label 21",
+					},
+				},
+				TotalCount: 20,
+			},
+			{
+				Items: []dataset.Option{
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 1",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 2",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 3",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 4",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 5",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 6",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 7",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 8",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 9",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 10",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 11",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 12",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 13",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 14",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 15",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 16",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 17",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 18",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 19",
+					},
+					{
+						DimensionID: "Dimension 2",
+						Label:       "Label 20",
+					},
+				},
+				TotalCount: 20,
+			},
+		}
+
+		Convey("when valid parameters are provided", func() {
+			p := CreateCensusDatasetLandingPage(
+				context.Background(),
+				req,
+				pageModel,
+				oneContactDetailDM,
+				versionOneDetails,
+				datasetOptions,
+				"",
+				false,
+				[]dataset.Version{},
+				1,
+				"",
+				"",
+				[]string{},
+				50,
+				false,
+				false,
+				filter.Model{Downloads: nil})
+			Convey("then truncation works as expected", func() {
```

Comment:
then the list should be truncated to show the first, middle, and last three values

Reply:
Will update the convey statement